src/WhereConditions/WhereHasConditionsDirective.php

```diff
@@ -43,7 +43,7 @@ public static function definition(): string
     /**
      * @param  \Illuminate\Database\Eloquent\Builder  $builder  The builder used to resolve the field.
      * @param  mixed  $whereConditions The client given conditions
-     * @return \Illuminate\Database\Eloquent\Builder  The modified builder.
+     * @return \Illuminate\Database\Eloquent\Builder | \Illuminate\Database\Query\Builder  The modified builder.
```

Comment:
I think this will not work with a non-eloquent query builder.

Reply:
I left that option on, in case someone would want to use that method manually, or from different place in code, but would only have a raw builder. If that shouldn't be the case, then yeah, it's unnecessary there. And it would break only if someone would try to use it manually and use `HAS`, which we can fix by checking if `model` before applying the `HAS`. Tell me which way you prefer.

Reply:
The Static Code Analysis complains if I remove it, because the `handleWhereConditions` returns both of builders.

Reply:
You can use a `/** @var */` PHPDoc to assert the result of `handleWhereConditions` is a specific type.